xgcm/test/test_integrate.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import print_function
+import pytest
+
+from xgcm.grid import Grid
+from xgcm.test.datasets import datasets_grid_metric
+
+
+@pytest.mark.parametrize("gridtype", ["B", "C"])
+def test_integrate(gridtype):
+    ds_full = datasets_grid_metric()
+    ds = ds_full[gridtype]
+    grid = Grid(
+        ds,
+        coords=ds_full["coords"],
+        metrics=ds_full["metrics"],
+        # periodic=True,
+    )
+    # test tracer position
+    for axis, metric_name, dim in zip(
+        ["X", "Y", "Z", ["X", "Y"], ["X", "Y", "Z"]],
+        ["dx_t", "dy_t", "dz_t", "area_t", "volume_t"],
+        ["xt", "yt", "zt", ["xt", "yt"], ["xt", "yt", "zt"]],
+    ):
+        integrated = grid.integrate(ds.tracer, axis)
+        expected = (ds.tracer * ds[metric_name]).sum(dim)
+        assert integrated.equals(expected)
+        # test with tuple input if list is provided
+        if isinstance(axis, list):
+            integrated = grid.integrate(ds.tracer, tuple(axis))
+            expected = (ds.tracer * ds[metric_name]).sum(dim)
+            assert integrated.equals(expected)
```

Comment:
This would be a place to test for the error of integrating over a dataset that doesn't have the requested axes.

Reply:
Thats a good point. Ill get onto that soon.